src/App.tsx

```diff
@@ -1,61 +1,139 @@
+  // #region inputs
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [userIdError, setUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const sanitizedValue = event.target.value.replace(
+      /[^a-zA-Zа-яА-ЯіІєЄїЇґҐ0-9\s]/g,
+      '',
+    );
+
+    setTitle(sanitizedValue);
+    setTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setUserIdError(false);
+  };
+  // #endregion
+
+  // #region todos
+  const [currentTodos, setCurrentTodos] = useState<Todo[]>([
+    ...todosFromServer,
+  ]);
+
+  const addTodo = (todo: Todo) => {
+    setCurrentTodos(todos => [...todos, todo]);
+  };
+
+  // #endregion
+
+  // #region return function
+  const clearInputs = () => {
+    setTitle('');
+    setUserId(0);
+    setTitleError(false);
+    setUserIdError(false);
+  };
+  // #endregion
+
+  // #region create id function
+  const uniqueId = () => {
+    const maxId = Math.max(...currentTodos.map(todo => todo.id));
+
+    return maxId + 1;
+  };
+  // #endregion
+
+  // #region add todo and check
+  const handleAddTodo = (event: React.FormEvent) => {
+    event.preventDefault();
+    setTitleError(!title);
+    setUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const createTodo: Todo = {
+      id: uniqueId(),
+      title,
+      completed: false,
+      userId,
+    };
+
+    addTodo(createTodo);
+
+    clearInputs();
+  };
+  // #endregion
+
+  const enrichedTodos = currentTodos.map(todo => ({
+    ...todo,
+    user: usersFromServer.find(user => user.id === todo.userId),
+  }));
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleAddTodo}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">Title: </label>
+          <input
+            type="text"
+            data-cy="titleInput"
+            id="title"
+            placeholder="Enter a title"
+            value={title}
+            onChange={handleTitleChange}
+          />
+          {titleError && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
+          <label htmlFor="select">User:</label>
+          <select
+            data-cy="userSelect"
+            id="select"
+            value={userId}
+            onChange={handleUserIdChange}
+          >
+            <option value="0">Choose a user</option>
+            {usersFromServer.map(user => (
```

Comment:
Directly mapping over `usersFromServer` is not recommended. Instead, pass the user data as props from a parent component to ensure a more maintainable and scalable architecture.